docs/reference/commandline/tag.md

```diff
@@ -16,5 +16,11 @@ parent = "smn_cli"
 
       --help               Print usage
 
+Image name must match the regular expression `[a-z0-9]+`. 
```

Comment:
I'd avoid any mention of `REGISTRYHOST` or `USERNAME`. These are interpretive conventions. https://github.com/docker/docker/pull/22665#discussion_r63022941 has the right format, slightly modified to this:

```
docker tag [OPTIONS] NAME[:TAG] NAME[:TAG]
```

Further documentation can clarify that the name may include a registry host, per this aspect of the grammar:

```
name                            := [hostname '/'] component ['/' component]*
```

As an aside, it seems like we should take the time to document that grammar directly. It should be broken down into the various roles:

1. Taggable names (or better word). These can be taken as user input and is the topic of discussion here.
2. The superset, known as references, including digests.

This is probably outside the scope of this PR.